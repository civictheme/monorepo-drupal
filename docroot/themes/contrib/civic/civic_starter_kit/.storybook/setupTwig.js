const twigDrupal = require('twig-drupal-filters');

/**
 * Configures and extends a standard twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
