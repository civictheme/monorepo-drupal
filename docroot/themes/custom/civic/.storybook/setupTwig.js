const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');

module.exports.namespaces = {
  atoms: resolve(__dirname, '../', 'components/01-atoms'),
  molecules: resolve(__dirname, '../', 'components/02-molecules'),
  organisms: resolve(__dirname, '../', 'components/03-organisms'),
  templates: resolve(__dirname, '../', 'components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
