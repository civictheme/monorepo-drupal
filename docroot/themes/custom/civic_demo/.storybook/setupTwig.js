const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');


module.exports.namespaces = {
  atoms: resolve(__dirname, '../', '.storybook-components/01-atoms'),
  molecules: resolve(__dirname, '../', '.storybook-components/02-molecules'),
  organisms: resolve(__dirname, '../', '.storybook-components/03-organisms'),
  templates: resolve(__dirname, '../', '.storybook-components/04-templates'),
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigDrupal(twig);
  return twig;
};
