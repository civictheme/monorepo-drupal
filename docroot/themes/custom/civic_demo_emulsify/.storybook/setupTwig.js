const { resolve } = require('path');
const twigDrupal = require('twig-drupal-filters');
const twigBEM = require('bem-twig-extension');
const twigAddAttributes = require('add-attributes-twig-extension');

module.exports.namespaces = {
  atoms: [
    resolve(__dirname, '../../civic_emulsify', 'components/01-atoms'),
    resolve(__dirname, '../', 'components/01-atoms'),
  ],
  molecules: [
    resolve(__dirname, '../../civic_emulsify', 'components/02-molecules'),
    resolve(__dirname, '../', 'components/02-molecules'),
  ],
  organisms: [
    resolve(__dirname, '../../civic_emulsify', 'components/03-organisms'),
    resolve(__dirname, '../', 'components/03-organisms'),
  ],
  templates: [
    resolve(__dirname, '../../civic_emulsify', 'components/04-templates'),
    resolve(__dirname, '../', 'components/04-templates'),
  ]
};

/**
 * Configures and extends a standard twig object.
 *
 * @param {Twig} twig - twig object that should be configured and extended.
 *
 * @returns {Twig} configured twig object.
 */
module.exports.setupTwig = function setupTwig(twig) {
  twig.cache();
  twigBEM(twig);
  twigDrupal(twig);
  twigAddAttributes(twig);
  return twig;
};
