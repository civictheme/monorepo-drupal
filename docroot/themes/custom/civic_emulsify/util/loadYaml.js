import { resolve } from 'path';
import { readFileSync } from 'fs';
import { parse } from 'yaml';
import R from 'ramda';

/**
 * Small utility function that loads a yaml file and parses it synchronously.
 * This is intended to make composition cleaner.
 *
 * @param {string} relativePath - relative path to a yaml file that will be loaded and parsed.
 *
 * @returns {string} JavaScript object that results from the yaml parsing of the specified file.
 */
export default function loadYaml(relativePath) {
  const fullPath = resolve(__dirname, relativePath);
  return parse(readFileSync(fullPath, 'utf8'));
}
