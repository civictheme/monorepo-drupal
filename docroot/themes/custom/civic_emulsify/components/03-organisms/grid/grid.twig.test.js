import { join } from 'path';
import { render, Twig } from 'twig-testing-library';

import loadYaml from '../../../util/loadYaml';
import { setupTwig, namespaces } from '../../../.storybook/setupTwig';

setupTwig(Twig);

describe('grid', () => {
  it('can render a grid', async () => {
    const { container } = await render(
      join(__dirname, 'grid.twig'),
      loadYaml(join(__dirname, 'grid.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });

  it('can render a grid of cards', async () => {
    const { container } = await render(
      join(__dirname, 'grid.twig'),
      loadYaml(join(__dirname, 'grid-cards.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });

  it('can render a grid of ctas', async () => {
    const { container } = await render(
      join(__dirname, 'grid.twig'),
      loadYaml(join(__dirname, 'grid-ctas.yml')),
      namespaces,
    );

    expect(container).toMatchSnapshot();
  });
});
