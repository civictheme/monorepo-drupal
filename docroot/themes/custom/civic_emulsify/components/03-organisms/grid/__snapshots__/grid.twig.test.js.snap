// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`grid can render a grid 1`] = `
<div>
  
  
  

  


  <h2
    class="h2"
  >
    
      Default
  
  </h2>
  

  <div
    class="grid grid--contained"
  >
    
              
    <div
      class="grid__item grid__item--example"
    />
    
            
    <div
      class="grid__item grid__item--example"
    />
    
            
    <div
      class="grid__item grid__item--example"
    />
    
      
  </div>
  

</div>
`;

exports[`grid can render a grid of cards 1`] = `
<div>
  
  

  
  


  <h2
    class="h2"
  >
    
      Card Grid
  
  </h2>
  

  <div
    class="grid grid--card"
  >
    
              


    <div
      class="card grid__item"
    >
      
  
            


  

      <img
        alt="People"
        class="image"
        src="https://placeimg.com/480/300/people"
      />
      
        
  
      <div
        class="card__content"
      >
        
    
          
      


        <h2
          class="card__heading"
        >
          
      


          <a
            class="card__heading-link"
            href="#"
          >
            
      Super Awesome Card
  
          </a>
          
  
        </h2>
        
      
        
          


        <h3
          class="card__subheading"
        >
          
      Person of Interest
  
        </h3>
        
        
          


        <p
          class="card__body"
        >
          
      Curabitur non nulla sit amet nisl tempus convallis quis ac lectus. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Pellentesque in ipsum id orci porta dapibus. Nulla quis lorem ut libero malesuada feugiat. Nulla quis lorem ut libero malesuada feugiat.
  
        </p>
        
        
          


        <a
          class="card__link"
          href="#"
        >
          
      Click here
  
        </a>
        
        
      
      </div>
      

    </div>
    
            


    <div
      class="card grid__item"
    >
      
  
            


  

      <img
        alt="Nature"
        class="image"
        src="https://placeimg.com/480/300/nature"
      />
      
        
  
      <div
        class="card__content"
      >
        
    
          
      


        <h2
          class="card__heading"
        >
          
      Card Number Two
  
        </h2>
        
      
        
          


        <h3
          class="card__subheading"
        >
          
      Some pretty sweet nature
  
        </h3>
        
        
          


        <p
          class="card__body"
        >
          
      Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat. Vivamus magna justo.
  
        </p>
        
        
        
      
      </div>
      

    </div>
    
            


    <div
      class="card grid__item"
    >
      
  
            


  

      <img
        alt="Animals"
        class="image"
        src="https://placeimg.com/480/300/animals"
      />
      
        
  
      <div
        class="card__content"
      >
        
    
          
      


        <h2
          class="card__heading"
        >
          
      The Last Card Heading
  
        </h2>
        
      
        
          


        <h3
          class="card__subheading"
        >
          
      Awesome animals
  
        </h3>
        
        
          


        <p
          class="card__body"
        >
          
      Tempus convallis quis ac lectus. Vivamus magna justo, lacinia eget consectetur sed, convallis at tellus. Donec rutrum congue leo eget malesuada. Proin eget tortor risus. Sed porttitor lectus nibh. Nulla quis lorem ut libero malesuada feugiat.. Nulla quis lorem ut libero malesuada feugiat.
  
        </p>
        
        
        
      
      </div>
      

    </div>
    
      
  </div>
  

</div>
`;

exports[`grid can render a grid of ctas 1`] = `
<div>
  
  


  


  <h2
    class="h2"
  >
    
      CTA Grid
  
  </h2>
  

  <div
    class="grid"
  >
    
              
    <div
      class="cta grid__item"
    >
      
  
      <h2>
        This is an awesome CTA
      </h2>
      
  




      <button
        class="button"
      >
        
      Click me!
  
      </button>
      

    </div>
    
            
    <div
      class="cta grid__item"
    >
      
  
      <h2>
        This is an even better CTA!!
      </h2>
      
  




      <button
        class="button"
      >
        
      No, click me!
  
      </button>
      

    </div>
    
            
    <div
      class="cta grid__item"
    >
      
  
      <h2>
        This is a boring CTA
      </h2>
      
  




      <button
        class="button"
      >
        
      Click me, I guess...
  
      </button>
      

    </div>
    
      
  </div>
  

</div>
`;
